src/Rules/Methods/CallToMethodStatementWithoutSideEffectsRule.php

```diff
@@ -61,8 +62,8 @@ public function processNode(Node $node, Scope $scope): array
 		}
 
 		$method = $calledOnType->getMethod($methodName, $scope);
-		if ($method->hasSideEffects()->no() || $node->expr->isFirstClassCallable()) {
-			if (!$node->expr->isFirstClassCallable()) {
+		if ($method->hasSideEffects()->no() || $methodCall->isFirstClassCallable()) {
+			if (!$methodCall->isFirstClassCallable()) {
 				$throwsType = $method->getThrowType();
```

Comment:
Also remove this part about throw type.

Reply:
IMHO this is why the test still fails.

Reply:
Oh this might be the reason it was still failing. Thanks